src/routes/DropDownButton.jsx

```diff
@@ -149,9 +149,10 @@ export default function CustomizedMenus() {
     // console.log(buttonText, " == ", "Create Pastebin")
     if(buttonText === "Encrypt to Pastebin") {
         console.log(text);
-        let res = encrypt(text)
-        //sets the pasteBinLink to output of usePasteBinPost
+        let res = await encrypt(text)
         console.log(res.data);
+        //sets the pasteBinLink to output of usePasteBinPost
+        await setPasteBinLink(res.data)
```

Comment:
@fairhurt How should I be calling the pastebin post function? I'm trying to test to see if its getting the api key from storage but I'm having a hard time making a request

Reply:
If you call setPostLink(res.data) it should automatically update the state and call the post function.

Reply:
Perfect, I was just over thinking it thanks